kubeflow/argo/argo.libsonnet

```diff
@@ -0,0 +1,320 @@
+                    name: "ARGO_NAMESPACE",
+                    valueFrom: {
+                      fieldRef: {
+                        apiVersion: "v1",
+                        fieldPath: "metadata.namespace",
+                      },
+                    },
+                  },
+                  {
+                    name: "IN_CLUSTER",
+                    value: "true",
+                  },
+                ],
+                image: "argoproj/argoui:v2.0.0-alpha3",
+                imagePullPolicy: "IfNotPresent",
+                name: "argo-ui",
+                resources: {},
+                terminationMessagePath: "/dev/termination-log",
+                terminationMessagePolicy: "File",
+              },
+            ],
+            dnsPolicy: "ClusterFirst",
+            restartPolicy: "Always",
+            schedulerName: "default-scheduler",
+            securityContext: {},
+            serviceAccount: "argo",
+            serviceAccountName: "argo",
+            terminationGracePeriodSeconds: 30,
+            readinessProbe: {
+              httpGet: {
+                path: "/",
+                port: 8001,
+              },
+            },
+          },
+        },
+      },
+    },  // deployUi
+
+    uiIngress:: {
+      apiVersion: "extensions/v1beta1",
+      kind: "Ingress",
+      metadata: {
+        name: "argo-ui",
+        namespace: namespace,
+      },
+      annotations: {
+        "kubernetes.io/ingress.global-static-ip-name": "argo-ui",
+      },
+      spec: {
+        rules: [
+          {
+            http: {
+              paths: [
+                {
+                  backend: {
+                    serviceName: "argo-ui",
+                    servicePort: 80,
+                  },
+                  path: "/*",
+                },
+              ],
+            },
+          },
+        ],
+      },
+    },  // ingress
+
+    uiService: {
+      apiVersion: "v1",
+      kind: "Service",
+      metadata: {
+        labels: {
+          app: "argo-ui",
+        },
+        name: "argo-ui",
+        namespace: namespace,
+      },
+      spec: {
+        ports: [
+          {
+            port: 80,
+            targetPort: 8001,
+          },
+        ],
+        selector: {
+          app: "argo-ui",
+        },
+        sessionAffinity: "None",
+        type: "NodePort",
+      },
+    },
+
+    config: {
+      apiVersion: "v1",
+      data: {
+        config: @"executorImage: argoproj/argoexec:v2.0.0-alpha2",
+      },
+      kind: "ConfigMap",
+      metadata: {
+        name: "workflow-controller-configmap",
+        namespace: namespace,
+      },
+    },
+
+    serviceAccount: {
+      apiVersion: "v1",
+      kind: "ServiceAccount",
+      metadata: {
+        name: "argo",
+        namespace: namespace,
+      },
+    },  // service account
+
+    // TODO(jlewi): Do we really need cluster admin privileges? Why?
+    // is this just because workflow controller is trying to create the CRD?
```

Comment:
In v2.0.0-beta1, the controller no longer attempts to create the CRD. This is left to installer now.

cluster-admin would be needed if workflow-controller needs to manage workflows across namespaces. If the controller is configured to only operate in a single namespace (this can be configured in the controller's configmap), then I think admin should be sufficient. But I haven't tried this yet.

We identified the minimum privileges required by the controller here:
https://github.com/argoproj/argo/blob/master/cmd/argo/commands/const.go#L20

Note that UI and controller have different set of privileges.

In 2.0.0-beta1, the installer will create a cluster-admin using these privileges. But if you would like to hand roll the deployment/rbac manifests yourself, argo 2.0.0-beta1 now has `--dry-run` flag to print the manifests without installing, and you can tweak it from there.

Reply:
Very helpful thanks. I updated to the beta1 version and locked down the permissions. PTAL.